portfolio/src/test/java/com/google/sps/CommentsServletGetTest.java

```diff
@@ -0,0 +1,158 @@
+        when(request.getParameter("type")).thenReturn("date");
+        when(request.getParameter("quantity")).thenReturn("all");
+
+        doGetRequest(request, response);
+        writer.flush();
+        String result = stringWriter.toString();
+
+        List<Map.Entry<Long, Comments.Comment>> expected =
+                Arrays.asList(new AbstractMap.SimpleEntry(
+                        commentEntity.getKey().getId(),
+                        Comments.Comment.getCommentFromEntity(commentEntity)));
+        assert (result.equals(gson.toJson(expected)));
+    }
+
+    @Test
+    public void testCommentOrderByDate() throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Date date = new Date();
+
+        Entity comment1 = new Entity(Comments.COMMENT_ENTITY_KIND);
+        comment1.setProperty(Comments.TEXT, "uno");
+        comment1.setProperty(Comments.RATING, 0L);
+        comment1.setProperty(Comments.DATE, date);
+        datastore.put(comment1);
+
+        date = new Date(date.getTime() + 1);
+        Entity comment2 = new Entity(Comments.COMMENT_ENTITY_KIND);
+        comment2.setProperty(Comments.TEXT, "dos");
+        comment2.setProperty(Comments.RATING, 0L);
+        comment2.setProperty(Comments.DATE, date);
+        datastore.put(comment2);
+
+        date = new Date(date.getTime() + 1);
+        Entity comment3 = new Entity(Comments.COMMENT_ENTITY_KIND);
+        comment3.setProperty(Comments.TEXT, "tres");
+        comment3.setProperty(Comments.RATING, 0L);
+        comment3.setProperty(Comments.DATE, date);
+        datastore.put(comment3);
+
+        date = new Date(date.getTime() + 1);
+        Entity comment4 = new Entity(Comments.COMMENT_ENTITY_KIND);
+        comment4.setProperty(Comments.TEXT, "cuatro");
+        comment4.setProperty(Comments.RATING, 0L);
+        comment4.setProperty(Comments.DATE, date);
+        datastore.put(comment4);
+
+        when(request.getParameter("type")).thenReturn("date");
+        when(request.getParameter("quantity")).thenReturn("3");
+
+        doGetRequest(request, response);
+        writer.flush();
+        String result = stringWriter.toString();
+
+        List<Map.Entry<Long, Comments.Comment>> expected =
+                Arrays.asList(
+                        new AbstractMap.SimpleEntry(
+                                comment4.getKey().getId(),
+                                Comments.Comment.getCommentFromEntity(comment4)),
+                        new AbstractMap.SimpleEntry(
+                                comment3.getKey().getId(),
+                                Comments.Comment.getCommentFromEntity(comment3)),
+                        new AbstractMap.SimpleEntry(
+                                comment2.getKey().getId(),
+                                Comments.Comment.getCommentFromEntity(comment2))
+                );
+        assert (result.equals(gson.toJson(expected)));
```

Comment:
You should use JUnit's `assertEquals(expected, result)` (and `assertTrue`, `assertNull`, etc., when applicable), or even `assertThat` (Truth or Hamcrest) if you prefer, instead of Java's assert.

Same applies to other tests